File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> us(nums.begin(), nums.end());
        int maxCount = 0;

        for(int num: us){
            if(!us.count(num-1)){
                int n = num;
                int count = 1;

                while(us.count(n+1)){
                    n++;
                    count++;
                }
                maxCount = max(maxCount, count);
            }
        }
        return maxCount;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)
        max_count = 0

        for num in num_set:
            if num-1 not in num_set:
                count = 1
                curr = num

                while curr+1 in num_set:
                    curr += 1
                    count += 1
                    
                max_count = max(count, max_count)
        
        return max_count